src/ml/neural_net/image_augmentation.cpp

```diff
@@ -117,5 +117,98 @@ image_augmenter::result resize_only_image_augmenter::prepare_images(
   return res;
 }
 
+image_augmenter::result processed_image_augmenter::prepare_images(
+    std::vector<labeled_image> source_batch) {
+  result res;
+
+  const size_t n = opts_.batch_size;
+  constexpr size_t c = 3;
+
+  std::vector<turi::neural_net::shared_float_array> images_to_aug;
+  std::vector<turi::neural_net::shared_float_array> ann_batch;
+  std::vector<turi::neural_net::shared_float_array> pred_batch;
+
+  // Decode a batch of images to raw format
+  for (size_t i = 0; i < n; i++) {
```

Comment:
What happens if `source_batch` does not contain exactly `n` elements?

Consider looking at what was done for `mps_image_augmenter` https://github.com/apple/turicreate/blob/4658fc57dba09f01806d5d5ad1a9c6bee670e916/src/ml/neural_net/mps_image_augmentation.mm#L332